pubsub.go

```diff
@@ -327,13 +327,19 @@ func NewPubSub(ctx context.Context, h host.Host, rt PubSubRouter, opts ...Option
 			h.SetStreamHandler(id, ps.handleNewStream)
 		}
 	}
-	h.Network().Notify((*PubSubNotif)(ps))
+
+	// start monitoring for new peers
+	notify := (*PubSubNotif)(ps)
+	if err := notify.startMonitoring(); err != nil {
+		return nil, fmt.Errorf("unable to start pubsub monitoring: %w", err)
+	}
 
 	ps.val.Start(ps)
 
 	go ps.processLoop(ctx)
 
-	(*PubSubNotif)(ps).Initialize()
+	// add current peers to notify system
+	notify.AddPeers(h.Network().Peers()...)
```

Comment:
ah ok, it is still here.

Reply:
This is racey; it needs to happen inside `startMonitoring`, after we have initialized the bus but _before_ we have spawned the monitoring goroutine.

Reply:
In the latest commit, the following test passes:

    go test -v -run=TestSimpleDiscovery -count=1 .

As you suggest, we could move `AddPeers` to inside `startMonitoring` before the goroutine as `p.AddPeers(p.host.Network().Peers()...)` . But with this change, `TestSimpleDiscovery` hangs and times out. What do you think?